rspotify-http/src/ureq.rs

```diff
@@ -1,15 +1,25 @@
 //! The client implementation for the ureq HTTP client, which is blocking.
 
-use super::{BaseHttpClient, Form, Headers, HttpError, HttpResult, Query};
+use super::{BaseHttpClient, Form, Headers, Query};
+
+use std::io;
 
 use maybe_async::sync_impl;
 use serde_json::Value;
 use ureq::{Request, Response};
 
-impl HttpError {
-    pub fn from_response(r: ureq::Response) -> Self {
-        HttpError::StatusCode(r.status(), r.status_text().to_string())
-    }
+/// Custom enum that contains all the possible errors that may occur when using
+/// `ureq`.
+#[derive(thiserror::Error, Debug)]
+pub enum UreqError {
+    #[error("transport: {0}")]
+    Transport(#[from] ureq::Transport),
+
+    #[error("I/O: {0}")]
+    Io(#[from] io::Error),
+
+    #[error("status code {}", ureq::Response::status(.0))]
```

Comment:
Would you like to add `///` doc to include a comment with a brief description?

Reply:
Will do